Add new KMS encryption feature, deprecate MINIO_ACCESS_KEY and MINIO_… (#341)
minio/docs: source/reference/minio-server/minio-server.rst

.. envvar:: MINIO_ACCESS_KEY_OLD

   .. deprecated:: RELEASE.2021-04-22T15-44-28Z

   To perform root credential rotation, modify the
   :envvar:`MINIO_ROOT_USER_FILE` and `MINIO_ROOT_PASSWORD_FILE` environment
   variables.

.. envvar:: MINIO_SECRET_KEY_OLD

   .. deprecated:: RELEASE.2021-04-22T15-44-28Z

   To perform root credential rotation, modify the
   :envvar:`MINIO_ROOT_USER_FILE` and `MINIO_ROOT_PASSWORD_FILE` environment
   variables.

> Key Management Service and Encryption

Author: @aead this is somewhat based on a very old KES PR I need to review, so let me know how far off base I am here.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. envvar:: MINIO_KMS_SECRET_KEY

   .. versionadded:: RELEASE.2021-04-22T15-44-28Z

>    The client-provided encryption key to use for encrypting the

Reviewer: Not just IAM data but also objects. MinIO uses the KMS to encrypt any backend data - IAM data / configuration by default. S3 objects (at the moment) only when requested.

   MinIO backend (users, groups, policies, and server configuration).
   See :minio-git:`KMS IAM/Config Encryption
   <minio/blob/master/docs/kms/IAM.md>` for more information.

   Prior to :minio-release:`RELEASE.2021-04-22T15-44-28Z`, MinIO used the
   :ref:`root <minio-users-root>` user credentials for encrypting the backend. 

.. envvar:: MINIO_KMS_KES_ENDPOINT

   The endpoint for the MinIO Key Encryption Service (KES) process to use
   for supporting SSE-S3 and MinIO backend encryption operations.

.. envvar:: MINIO_KMS_KES_KEY_FILE

   The private key associated to the the :envvar:`MINIO_KMS_KES_CERT_FILE` 
   x.509 certificate to use when authenticating to the KES server. 

>    The KES server requires clients to present both their certificate and
>    private key for performing mutual TLS (mTLS).

Reviewer: ```suggestion
   The KES server requires clients to present their certificate for performing mutual TLS (mTLS).
```

   See the :minio-git:`KES wiki <kes/wiki/Configuration#policy-configuration>` 
   for more complete documentation on KES access control.

.. envvar:: MINIO_KMS_KES_CERT_FILE

   The x.509 certificate to present to the KES server. 

>    The KES server requires clients to present both their certificate and
>    private key for performing mutual TLS (mTLS).

Reviewer: ```suggestion
   The KES server requires clients to present their certificate and for performing mutual TLS (mTLS).
```

   The KES server computes an 
   :minio-git:`identity <kes/wiki/Configuration#policy-configuration>`
   from the certificate and compares it to its configured    
   policies. The KES server grants the
   :mc:`minio` server access to only those operations explicitly granted by the
   policy.

   See the :minio-git:`KES wiki <kes/wiki/Configuration#policy-configuration>`
   for more complete documentation on KES access control.

.. envvar:: MINIO_KMS_KES_KEY_NAME

>    The name of an external ke to retrieve from the Key Management System (KMS)

Reviewer: ```suggestion
   The name of an external key at the Key Management System (KMS) to perform en/decryption operations
```
